Makefile

```diff
@@ -6,9 +6,25 @@
 IMAGE_PREFIX ?= quay.io/cortexproject/
 # Use CIRCLE_TAG if present for releases.
 IMAGE_TAG ?= $(if $(CIRCLE_TAG),$(CIRCLE_TAG),$(shell ./tools/image-tag))
-GIT_REVISION := $(shell git rev-parse HEAD)
 UPTODATE := .uptodate
 
+# Boiler plate for building Cortex's Golang binary.
+VERSION := $(shell cat VERSION)
+GIT_BRANCH := $(shell ./tools/image-tag)
+GIT_REVISION := $(shell git rev-parse HEAD)
+GO_LDFLAGS := -X github.com/prometheus/common/version.Version=$(VERSION) \
```

Comment:
Note that this will break in Go 1.14, which will use vendor directory by default, if present and go.mod specifies go 1.14. Vendored dependencies need to be specified as `github.com/cortexproject/cortex/vendor/github.com/prometheus/common/version` :-(

I think easiest solution to handle both scenarios is to have simple build.go in the Cortex project, that will then set Prometheus values in `init` method.